File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/SpringBootConfigurationEnvironmentSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableSpringBootConfigurationEnvironmentSettings.class)
public interface SpringBootConfigurationEnvironmentSettings {
  Optional<String> vmOptions();

  Optional<Boolean> includeDependenciesWithProvidedScope();

  Optional<String> useClassPathOfModule();

}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/configurations/spring_boot/SpringBootConfigurationEnvironmentSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.configurations.spring_boot

import com.fasterxml.jackson.annotation.JsonProperty

data class SpringBootConfigurationEnvironmentSettings(

    @JsonProperty("VM options")
    val vmOptions: String? = null,

    @JsonProperty("Include dependencies with provided scope")
    val includeDependenciesWithProvidedScope: Boolean? = null,

    @JsonProperty("Use classpath of module")
    val useClasspathOfModule: String? = null
)
